backend/database/utils/dynamic_field_manager.py

```diff
@@ -73,22 +77,58 @@ def get_db_proto_union(proto_field_list: List[ProtoFieldResult], db_object: DBOb
     return result_list
 
 
-def create_and_filter_proto_field(proto_message: message, blacklist_field_names: List[str],
-                                  blacklist_message_types: List[str], db_object: DBObjectBase) -> List[ProtoFieldResult]:
+def get_whitelist_proto_union(proto_field_list: List[ProtoFieldResult],
+                              whitelist_field_name: [str],
+                              whitelist_field_type: [str]) -> List[ProtoFieldResult]:
+    result_list = []
+    allowed = False
+    for item in proto_field_list:
+        if whitelist_field_name is not None and item.field_name in whitelist_field_name:
```

Comment:
This can all be nested under `if whitelist_field_name is not None`

Reply:
What do you mean by nested?

Reply:
```suggestion
        if whitelist_field_name is not None:
             if item.field_name in whitelist_field_name:
                 allowed = True
             for field_type in whitelist_field_type:
                 if field_type in item.nested_parents:
                     allowed = True
                 
```

Reply:
Of course you'd remove the lines after but idk how to do that with a suggestion